core/src/main/java/org/apache/accumulo/core/iterators/SynchronizedServerFilter.java

```diff
@@ -48,13 +49,19 @@ public SynchronizedServerFilter(SortedKeyValueIterator<Key,Value> source) {
 
   @Override
   public synchronized void next() throws IOException {
+    if (Thread.interrupted()) {
```

Comment:
I'm not sure what the penalty is for calling this for every K/V. I'm also wondering if the message should be logged as well in case the RuntimeException is ignored in a higher level iterator.

Reply:
Looking at NativeMap and SortedMapIterator, it appears the pattern is to check the interrupt on every seek and every 100th next.

Reply:
> Looking at NativeMap and SortedMapIterator, it appears the pattern is to check the interrupt on every seek and every 100th next.

What about checking the thread interrupt status in those places and in RFile?  That same places where the boolean is checked, those places are meant to run less frequently.  A long long time ago I did performance test and could see differences in the number of key values per second read depending on how frequently the volatile was checked (suspected it was interfering w/ the processor cache, but was not sure).  The check of the atomic boolean in RFile is done when a new data block is read so that is naturally less frequent.  NativeMap code does the periodic check thing you mentioned.

Reply:
Added in f02d89a.